tests/bsim/bluetooth/mesh/src/test_advertiser.c

```diff
@@ -580,6 +580,189 @@ static void test_rx_random_order(void)
+		if (adv_data->i == CONFIG_BT_MESH_ADV_BUF_COUNT - 1) {
+			k_sem_give(&adv_suspended_sem);
+		}
+	} else {
+		ASSERT_EQUAL(err, 0);
+	}
+}
+
+static void adv_create_and_send(bool suspend, uint8_t first_byte, struct adv_suspend_ctx *adv_data)
+{
+	struct bt_mesh_adv *advs[CONFIG_BT_MESH_ADV_BUF_COUNT];
+	static const struct bt_mesh_send_cb send_cb = {
+		.start = adv_send_start,
+		.end = adv_send_end,
+	};
+
+	for (int i = 0; i < CONFIG_BT_MESH_ADV_BUF_COUNT; i++) {
+		adv_data[i].suspend = suspend;
+		adv_data[i].i = i;
+
+		advs[i] = bt_mesh_adv_create(BT_MESH_ADV_DATA, BT_MESH_ADV_TAG_LOCAL,
+					    BT_MESH_TRANSMIT(2, 20), K_NO_WAIT);
+		ASSERT_FALSE_MSG(!advs[i], "Out of advs\n");
+
+		net_buf_simple_add_u8(&advs[i]->b, first_byte);
+		net_buf_simple_add_u8(&advs[i]->b, i);
+	}
+
+	for (int i = 0; i < CONFIG_BT_MESH_ADV_BUF_COUNT; i++) {
+		bt_mesh_adv_send(advs[i], &send_cb, &adv_data[i]);
+		bt_mesh_adv_unref(advs[i]);
+	}
+}
+
+static void test_tx_disable(void)
+{
+	struct adv_suspend_ctx adv_data[CONFIG_BT_MESH_ADV_BUF_COUNT];
+	struct bt_mesh_adv *extra_adv;
+	int err;
+
+	bt_init();
+	adv_init();
+
+	/* Fill up the adv pool and suspend the advertiser in the first start callback call. */
+	adv_create_and_send(true, 0xAA, adv_data);
+
+	err = k_sem_take(&adv_suspended_sem, K_SECONDS(10));
+	ASSERT_OK_MSG(err, "Not all advs were sent");
+
+	extra_adv = bt_mesh_adv_create(BT_MESH_ADV_DATA, BT_MESH_ADV_TAG_LOCAL,
+				       BT_MESH_TRANSMIT(2, 20), K_NO_WAIT);
+	ASSERT_TRUE_MSG(!extra_adv, "Created adv while suspended");
+
+	adv_resume();
+
+	/* Fill up the adv pool and suspend the advertiser and let it send all advs. */
+	adv_create_and_send(false, 0xBB, adv_data);
+
+	err = k_sem_take(&adv_sent_sem, K_SECONDS(10));
+	ASSERT_OK_MSG(err, "Not all advs were sent");
+
+	PASS();
+}
+
+static void suspended_adv_scan_cb(const bt_addr_le_t *addr, int8_t rssi, uint8_t adv_type,
+				  struct net_buf_simple *buf)
+{
+	uint8_t length;
+	uint8_t type;
+	uint8_t pdu;
+
+	if (adv_type != BT_GAP_ADV_TYPE_ADV_NONCONN_IND) {
+		return;
+	}
+
+	length = net_buf_simple_pull_u8(buf);
+	ASSERT_EQUAL(buf->len, length);
+	ASSERT_EQUAL(length, sizeof(uint8_t) * 3);
+	type = net_buf_simple_pull_u8(buf);
+	ASSERT_EQUAL(BT_DATA_MESH_MESSAGE, type);
+
+	pdu = net_buf_simple_pull_u8(buf);
+	if (pdu == 0xAA) {
+		pdu = net_buf_simple_pull_u8(buf);
+
+		/* Because the advertiser is stopped after the advertisement has been passed to the
+		 * host, the controller could already start sending the message. Therefore, if the
+		 * tester receives an advertisement with the first byte as 0xAA, the second byte can
+		 * only be 0x00. This applies to both advertisers.
+		 */
+		ASSERT_EQUAL(0, pdu);
+	}
+}
+
+static void test_rx_disable(void)
+{
+	int err;
+
+	bt_init();
+
+	/* It is sufficient to check that the advertiser didn't sent PDUs which the end callback was
+	 * not called for.
+	 */
+	err = bt_mesh_test_wait_for_packet(suspended_adv_scan_cb, &observer_sem, 20);
```

Comment:
```suggestion
	ASSERT_EQUAL(-ETIMEDOUT, bt_mesh_test_wait_for_packet(suspended_adv_scan_cb, &observer_sem, 20));
```
The first argument value is `expected` the second one is `gotten`. No logic impact, but error message considers that.

Reply:
Ok, but function can't be put into `ASSERT_EQUAL` macro:
https://github.com/zephyrproject-rtos/zephyr/blob/3c25bfefabaf2ab2a56e208e7372f62373a9a376/tests/bsim/bluetooth/mesh/src/mesh_test.h#L107-L115